nucypher/policy/conditions/types.py

```diff
@@ -1,6 +1,66 @@
-from typing import Dict, List, Union
+import sys
 
-ConditionValues = Union[str, int, bool]
-ReturnValueDict = Dict[str, ConditionValues]
-ConditionDict = Dict[str, Union[ConditionValues, ReturnValueDict]]
-LingoList = List[ConditionDict]
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+from typing import Any, Dict, List, Type, Union
+
+from web3.types import ABIFunction
+
+#########
+# Context
+#########
+ContextDict = Dict[str, Any]
+
+
+################
+# ConditionLingo
+################
+
+#
+# OperatorDict represents:
+# - {"operator": "and" | "or"}
+class OperatorDict(TypedDict):
+    operator: str
+
+
+#
+# ConditionDict is a dictionary of:
+# - str -> Simple values (str, int, bool), or parameter list which can be anything ('Any')
+# - "returnValueTest" -> Return Value Test definitions
+# - "functionAbi" -> ABI function definitions (already defined by web3)
+#
+BaseValue = Union[str, int, bool]
+MethodParameters = List[Any]
+
+ConditionValue = Union[BaseValue, MethodParameters]  # base value or list of base values
+
+
+# Return Value Test
+class ReturnValueTestDict(TypedDict, total=False):
+    comparator: str
+    value: Any
+    key: Union[str, int]
+
+
+ConditionDict = Dict[str, Union[ConditionValue, ReturnValueTestDict, ABIFunction]]
```

Comment:
Another good candidate for a `TypedDict`

Reply:
I'm open to ideas. This one is more abstract because of the different possible values for Condtions - unless you mean definining all Conditions here as well (RPC, Contract, Time)...